packages/plugin-react/src/index.ts

```diff
@@ -261,11 +269,21 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
             // Even if the automatic JSX runtime is not used, we can still
             // inject the React import for .jsx and .tsx modules.
             if (!skipReactImport && !importReactRE.test(code)) {
-              code = `import React from 'react'; ` + code
+              prependReactImport = true
             }
           }
         }
 
+        let inputMap: SourceMap | undefined
+        if (prependReactImport) {
+          code = prependReactImportCode + code
+          if (needHiresSourcemap) {
+            const s = new MagicString(code)
+            s.prepend(prependReactImportCode)
```

Comment:
Don't `s` ends up with two `prependReactImportCode` prepended here?
Seems that 279 should be in an else branch for the 280 condition

Reply:
Good catch.
I shouldn't have move `code = prependReactImportCode + code` before `if` without thinking of the effect... 🤦‍♂️